algorithms/dp/longest_common_subsequence.py

```diff
@@ -0,0 +1,22 @@
+def lcs(X , Y): 
+    
+    l1 = len(X) 
+    l2 = len(Y) 
+  
+    dp = [[0]*(l2+1) for i in xrange(l1+1)] 
+  
+    for i in range(l1+1): 
+        for j in range(l2+1): 
+            if i == 0 or j == 0 : 
+                dp[i][j] = 0
+            elif X[i-1] == Y[j-1]: 
+                dp[i][j] = dp[i-1][j-1]+1
+            else: 
+                dp[i][j] = max(dp[i-1][j] , dp[i][j-1]) 
+  
+    return dp[l1][l2]  
+
+
+X = "YRHHYDKP"
```

Comment:
can you add a unittest ?
Please take a look at [math unit tests](https://github.com/keon/algorithms/blob/master/tests/test_maths.py).
You can create a new file called `test_dp.py` for this as we have not created one yet.